server/internal/core/application/utils.go

```diff
@@ -20,8 +21,9 @@ type timedPayment struct {
 }
 
 type paymentsMap struct {
-	lock     *sync.RWMutex
-	payments map[string]*timedPayment
+	lock          *sync.RWMutex
+	payments      map[string]*timedPayment
+	ephemeralKeys map[string]*secp256k1.PublicKey
```

Comment:
either create a clone of `paymentsMap` but for pubkeys, or we add another lock here to control access to the ephemeralKeys map.

Using the same lock can lead to unnecessary and unexpected waiting time for a sender to register his pubkeys in a scenario where there are a lot of users registering their inputs at nearly the same time